pages/components/PlayerInfo.js

```diff
@@ -0,0 +1,47 @@
+import {useState, useEffect} from 'react'
+import styles from '@/styles/Home.module.css'
+import SeasonSelector from './SeasonSelector'
+ export default function PlayerInfo() {
+    const [player, setPlayer] = useState('')
+    const [loading, setLoading] = useState(true)
+    const [season, setSeason] = useState('2022')
+
+    useEffect(() => {
+    getPlayer(season)
+  }, [season])
+
+//Request functions
+  async function getPlayer(season){
+    const response = await fetch(`api/getLebron?season=${season}`)
+    const data = await response.json()
+    setPlayer(data)
+    setLoading(false)
+  }
+
+    const handleGetSeason = () => {
+    setSeason('')
+  }
+  return (
+    <>
+    <SeasonSelector onChange={setSeason} />
+    <div className={styles.playerText}>
+        <div className={styles.playerProfile}>
+            {/* {loading && <p>Loading...</p>} */}
+            <p>Name: {player.firstName} {player.lastName}</p>
+            <p>Position: {player.position}</p>
+            {loading ? <p>Loading Team...</p> : <p>Team: {player.team}</p>}
+            {/* if loading show loading text and if else show player text */}
+        </div>
+        <br />
+        <ul className={styles.playerStats}>
+            <li>Points: {player.points}</li>
+            <li>Assists: {player.assists}</li>
+            <li>Rebounds: {player.rebounds}</li>
+            <li>Blocks: {player.blocks}</li>
+            <li>Steals: {player.steals}</li>
+        </ul>
+    </div>
```

Comment:
This section is the what I meant should be pulled out into a component. That way we can eventually have 

```jsx
<SeasonSelector />
{loading ? <Loading /> : <PlayerData />}
```

I don't think there's anything wrong with what you did. I just think this will allow us a way to move forward

Reply:
@aisflat439 I had pulled just that section of HTML markup out but errors were thrown for `player.*` seemingly because the functions were not also included in the functions.

I pulled the useState and other related functions into the PlayerData file and those errors went away but then the data would not update when a choice was made on the `<SeasonSelector />` component.  For that reason, I moved that component and everything it needs to work into PlayerData as well. Can you show me a better way to accomplish this?

Reply:
It's likely that you passed props incorrectly. You want all the logic stuff, fetching, useState etc... Up top in the page. Then, pass down the state into the components. 

I'll make some more accurate suggestions when I get home later today but give that a go.

Reply:
Here's an article about passing props that's pretty great

https://beta.reactjs.org/learn#sharing-data-between-components